ibm/resource_ibm_is_instance.go

```diff
@@ -1680,6 +1704,57 @@ func instanceUpdate(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 	id := d.Id()
+
+	if d.HasChange(isInstanceAction) && !d.IsNewResource() {
+
+		_, actiontypeIntf := d.GetChange(isInstanceAction)
+		actiontype := actiontypeIntf.(string)
+		if actiontype != "" {
+			getinsOptions := &vpcv1.GetInstanceOptions{
+				ID: &id,
+			}
+			instance, response, err := instanceC.GetInstance(getinsOptions)
+			if err != nil {
+				if response != nil && response.StatusCode == 404 {
+					d.SetId("")
```

Comment:
@hkantare , should we just throw an error instead of setting id to "" in the update method..

Reply:
should we keep it here ? i see "" is set in update method within other attributes change as well